jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/internal/HttpConnectionOverHTTP.java

```diff
@@ -159,12 +162,40 @@ public SendFailure send(HttpExchange exchange)
         return delegate.send(exchange);
     }
 
+    public boolean isInitialize()
+    {
+        return initialize;
+    }
+
+    public void setInitialize(boolean initialize)
+    {
+        this.initialize = initialize;
+    }
+
     @Override
     public void onOpen()
     {
         super.onOpen();
         fillInterested();
-        promise.succeeded(this);
+        boolean initialize = isInitialize();
+        if (initialize)
+        {
+            Destination destination = getHttpDestination();
+            Request request = destination.getHttpClient().newRequest(destination.getOrigin().asString())
+                .method(HttpMethod.OPTIONS)
+                .path("*");
```

Comment:
Maybe put this into a protected method, so that it is easy to create a different request for initialization

Reply:
`HttpConnectionOverHTTP` is internal, so we don't want to promote overriding.

I'd say we will address the need to customize this request in the future, if necessary.